tests/models/pop2piano/test_tokenization_pop2piano.py

```diff
@@ -0,0 +1,159 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Please note that Pop2PianoTokenizer is too far from our usual tokenizers and thus cannot use the TokenizerTesterMixin class.
+"""
+
+import unittest
+
+import numpy as np
+from datasets import load_dataset
+
+from transformers.feature_extraction_utils import BatchFeature
+from transformers.testing_utils import (
+    is_pretty_midi_available,
+    is_torch_available,
+    require_pretty_midi,
+    require_torch,
+    slow,
+)
+from transformers.tokenization_utils import BatchEncoding
+
+
+if is_torch_available():
+    import torch
+
+    from transformers.models.pop2piano.modeling_pop2piano import Pop2PianoForConditionalGeneration
+
+requirements_available = is_torch_available() and is_pretty_midi_available()
+if requirements_available:
+    import pretty_midi
+
+    from transformers import Pop2PianoFeatureExtractor, Pop2PianoTokenizer
+
+
+## TODO : changing checkpoints from `susnato/pop2piano_dev` to `sweetcocoa/pop2piano` after the PR is approved
+
+
+@require_torch
+@require_pretty_midi
+class Pop2PianoTokenizerTest(unittest.TestCase):
```

Comment:
> There should also be tests for this tokenizer's specific logic e.g. mapping to / from the token_type and values

We already have those tests in `test_call` and `test_batch_decode`.

In addition to that, I have added the following tests - `test_get_vocab`, `test_save_and_load_tokenizer`, `test_pickle_tokenizer`, `test_padding_side_in_kwargs`, `test_truncation_side_in_kwargs`, `test_right_and_left_padding`, `test_right_and_left_truncation`, `test_padding_to_multiple_of` and `test_padding_with_attention_mask`.

Btw `test_internal_consistency` is not valid for this tokenizer because the outputs of the Feature Extractor is used in the tokenizer's `batch_decode` method in order to influence the conversion of the `token_ids` to `notes`, whereas the `__call__` method converts `notes` to `token_ids` without using/generating similar outputs like the feature extractor.

Reply:
> If I am not wrong you are talking about conversion between token_ids and tokens(notes here) right? We already have that in the tests - test_call and test_batch_decode.

Yes, that's what I'm talking about. AFAICT, these tests only test the type of the returned objects, not their values. It would be good to have at least one test that makes sure the correct token type is mapped to notes in the `relative_tokens_ids_to_notes` and `notes_to_midi` methods

Reply:
Hi @amyeroberts, in the recent push, I have extended the `test_call` to test for accuracy of the outputs of `__call__` method and created another test (`test_batch_decode_outputs`) to check for the accuracy of the outputs of `batch_decode` method.  

Please let me know if any more tests are needed or not.